workers/unity/Packages/com.improbable.gdk.core/Worker/ViewDiff.cs

```diff
@@ -0,0 +1,305 @@
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var storage))
+            {
+                throw new ArgumentException(
+                    $"Can not set remove component with ID {componentId} from entity with ID {entityId}. " +
+                    $"Unknown component ID");
+            }
+
+            storage.RemoveEntityComponent(entityId);
+        }
+
+        public void SetAuthority(long entityId, uint componentId, Authority authority)
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var authorityStorage))
+            {
+                throw new ArgumentException(
+                    $"Can not set authority over component with ID {componentId} for entity with ID {entityId}. " +
+                    $"Unknown component ID");
+            }
+
+            ((IDiffAuthorityStorage) authorityStorage).AddAuthorityChange(
+                new AuthorityChangeReceived(authority, new EntityId(entityId)));
+
+            // Remove received command requests if authority has been lost
+            if (authority == Authority.NotAuthoritative)
+            {
+                if (componentIdToCommandIdToStorage.TryGetValue(componentId, out var commandIdToStorage))
+                {
+                    foreach (var storage in commandIdToStorage)
+                    {
+                        storage.Value.RemoveEntityComponent(entityId);
+                    }
+                }
+            }
+        }
+
+        public void AddComponentUpdate<T>(T update, long entityId, uint componentId, uint updateId)
+            where T : ISpatialComponentUpdate
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var storage))
+            {
+                throw new ArgumentException(
+                    $"Can not update component with ID {componentId} on entity with ID {entityId}. " +
+                    $"Unknown component ID");
+            }
+
+            ((IDiffUpdateStorage<T>) storage).AddUpdate(new ComponentUpdateReceived<T>(update, new EntityId(entityId),
+                updateId));
+        }
+
+        public void AddEvent<T>(T ev, long entityId, uint componentId, uint updateId) where T : IEvent
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var storage))
+            {
+                throw new ArgumentException(
+                    $"Can not add event from component with ID {componentId} on entity with ID {entityId}. " +
+                    $"Unknown component ID");
+            }
+
+            ((IDiffEventStorage<T>) storage).AddEvent(new ComponentEventReceived<T>(ev, new EntityId(entityId),
+                updateId));
+        }
+
+        public void AddCommandRequest<T>(T request, uint componentId, uint commandId) where T : IReceivedCommandRequest
+        {
+            var storage = componentIdToCommandIdToStorage[componentId][commandId];
```

Comment:
shouldn't we do a "does this component id exist" check for all the command methods as well?

Reply:
Added

Reply:
Just forgot to add it to these ones.